framework/src/functions/PiecewiseConstant.C

```diff
@@ -8,85 +8,73 @@
 //* https://www.gnu.org/licenses/lgpl-2.1.html
 
 #include "PiecewiseConstant.h"
+#include "MathUtils.h"
 
 registerMooseObject("MooseApp", PiecewiseConstant);
 
-defineLegacyParams(PiecewiseConstant);
-
 InputParameters
 PiecewiseConstant::validParams()
 {
   InputParameters params = PiecewiseTabularBase::validParams();
-  MooseEnum direction("left right", "left");
+  MooseEnum direction("LEFT RIGHT LEFT_INCLUSIVE RIGHT_INCLUSIVE", "LEFT");
   params.addParam<MooseEnum>(
       "direction", direction, "Direction to look to find value: " + direction.getRawNames());
   params.addClassDescription("Defines data using a set of x-y data pairs");
   return params;
 }
 
-PiecewiseConstant::DirectionEnum
-PiecewiseConstant::getDirection(const std::string & direction)
-{
-  DirectionEnum dir = PiecewiseConstant::UNDEFINED;
-  if (direction == "left")
-  {
-    dir = PiecewiseConstant::LEFT;
-  }
-  else if (direction == "right")
-  {
-    dir = PiecewiseConstant::RIGHT;
-  }
-  else
-  {
-    mooseError("Unknown direction in PiecewiseConstant");
-  }
-  return dir;
-}
-
 PiecewiseConstant::PiecewiseConstant(const InputParameters & parameters)
-  : PiecewiseTabularBase(parameters), _direction(getDirection(getParam<MooseEnum>("direction")))
+  : PiecewiseTabularBase(parameters),
+    _direction(getParam<MooseEnum>("direction").template getEnum<DirectionEnum>())
```

Comment:
```suggestion
    _direction(getParam<MooseEnum>("direction").getEnum<DirectionEnum>())
```

this is not a template dependent name